internal/aad/aad_test.go

```diff
@@ -28,6 +28,8 @@ func TestAuthenticate(t *testing.T) {
 		"invalid server response":    {username: "invalid server response", wantErr: aad.ErrDeny},
 		"invalid credentials":        {username: "invalid credentials", wantErr: aad.ErrDeny},
 		"no such user":               {username: "no such user", wantErr: aad.ErrDeny},
+		"public client disallowed":   {appID: "public client disallowed", wantErr: aad.ErrDeny},
+		"no tenant-wide consent":     {appID: "no tenant-wide consent", wantErr: aad.ErrDeny},
```

Comment:
Would you mind adding those right after the `"can't connect to authority"` case? Since those error cases are related to external errors, it's better to have them grouped together.